src/Instance/MoveHoldingContext/MoveHoldingContext.test.js

```diff
@@ -0,0 +1,201 @@
+      instanceRelationshipTypesById: keyBy(identifierTypes, 'id'),
+      instanceFormats: [],
+      modesOfIssuance: [],
+      natureOfContentTerms: [],
+      tagsRecords: [],
+      locationsById,
+    }}
+    >
+      <MoveHoldingContext
+        leftInstance={leftInstance}
+        rightInstance={rightInstance}
+        moveHoldings={moveHoldings}
+      >
+        <InstanceMovementDetailsContainer
+          instance={leftInstance}
+          onClose={onClose}
+          data-test-movement-from-instance-details
+          id="movement-from-instance-details"
+        />
+
+        <InstanceMovementDetailsContainer
+          instance={rightInstance}
+          onClose={onClose}
+          data-test-movement-to-instance-details
+          id="movement-to-instance-details"
+        />
+      </MoveHoldingContext>
+    </DataContext.Provider>
+  </Router>,
+  translationsProperties
+);
+
+
+describe('MoveHoldingContext', () => {
+  it('should render correct holdings accordion for left pane', async () => {
+    const { getByTestId } = renderMoveHoldingContext();
+
+    const leftPane = getByTestId('movement-from-instance-details');
+    expect(within(leftPane).getByText(/Holdings: Main Library/)).toBeInTheDocument();
+    expect(within(leftPane).getByText(/Holdings: Annex/)).toBeInTheDocument();
+  });
+
+  it('should render "Drop holding" area in right pane', () => {
+    const { getByTestId } = renderMoveHoldingContext();
+
+    const leftPane = getByTestId('movement-to-instance-details');
+    expect(within(leftPane).getByText(/Drop holding/)).toBeInTheDocument();
+  });
+
+  it('"Select holdings" checkbox functionality works as expected', async () => {
+    const { getByTestId } = renderMoveHoldingContext();
+
+    const selectHoldingCheckbox = getByTestId('select-holding-c4a15834-0184-4a6f-9c0c-0ca5bad8286d');
+
+    fireEvent.click(selectHoldingCheckbox);
+    expect(selectHoldingCheckbox).toBeChecked();
+
+    fireEvent.click(selectHoldingCheckbox);
+    expect(selectHoldingCheckbox).not.toBeChecked();
+  });
+
+  it('should render confirmation modal with initial state in background', async () => {
+    const { getByText } = renderMoveHoldingContext();
+
+    expect(getByText(/ConfirmationModal/)).toBeInTheDocument();
+    expect(getByText(/0 items will be moved to/)).toBeInTheDocument();
+  });
+
+  it('should render correct list of holdings for Annex with checkbox', async () => {
+    const { getByTestId } = renderMoveHoldingContext();
+
+    const annexSection = getByTestId('c4a15834-0184-4a6f-9c0c-0ca5bad8286d');
+    const annexHoldingsAccordionBtn = getByTestId('accordion-toggle-button-c4a15834-0184-4a6f-9c0c-0ca5bad8286d');
+
+    fireEvent.click(annexHoldingsAccordionBtn);
+
+    let grid;
+    let rows;
+
+    await waitFor(() => {
+      grid = within(annexSection).getByRole('grid');
+      rows = within(grid).getAllByRole('row');
+
+      expect(grid).toBeVisible();
+      expect(rows).toHaveLength(3);
+    });
```

Comment:
Maybe it would be better to check by Holdings location names for example rather than by tags as far as we testing UI?